rust-runtime/aws-smithy-runtime/src/client/retries/strategy/standard.rs

```diff
@@ -210,8 +210,11 @@ impl RetryStrategy for StandardRetryStrategy {
             .unwrap_or(false);
         update_rate_limiter_if_exists(runtime_components, cfg, is_throttling_error);
 
-        // on success release any retry quota held by previous attempts
+        // on success release any retry quota held by previous attempts and award success tokens
         if !ctx.is_failed() {
+            // When a request succeeds, we grant an award, if present
+            token_bucket.reward_success();
+
             if let NoPermitWasReleased = self.release_retry_permit() {
```

Comment:
Do we need to worry about [`add_permits`](https://docs.rs/tokio/latest/tokio/sync/struct.Semaphore.html#method.add_permits) panicking if we exceed `MAX_PERMITS`? e.g. since we are now rewarding on success (or I believe eventually you want to refill on time as well) then on `Drop` a released permit puts it's tokens back.